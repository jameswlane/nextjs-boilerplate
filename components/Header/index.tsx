import React, { ReactElement } from 'react'
import Link from 'next/link'
import { useRouter } from 'next/router'

const Header = (): ReactElement => {
  const router = useRouter()
  const isPage = (path: string): boolean => path === router.pathname

  return (
    <header>
      <nav className="navbar navbar-expand-md navbar-dark fixed-top bg-dark">
        <Link href="/">
          <a className="navbar-brand" data-testid="top-nav-brand">
            Next.js Boilerplate
          </a>
        </Link>
        <div className="collapse navbar-collapse">
          <ul className="navbar-nav mr-auto">
            <li className={isPage('/') ? 'nav-item active' : 'nav-item'}>
              <Link href="/">
                <a className="nav-link" data-testid="top-nav-home">
                  Home
                </a>
              </Link>
            </li>
            <li className={isPage('/about') ? 'nav-item active' : 'nav-item'}>
              <Link href="/about">
                <a className="nav-link" data-testid="top-nav-about">
                  About
                </a>
              </Link>
            </li>
            <li className={isPage('/contact') ? 'nav-item active' : 'nav-item'}>
              <Link href="/contact">
                <a className="nav-link" data-testid="top-nav-contact">
                  Contact
                </a>
              </Link>
            </li>
          </ul>
        </div>
      </nav>
    </header>
  )
}

export default Header
