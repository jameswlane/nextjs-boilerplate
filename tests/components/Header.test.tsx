import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import Header from '../../components/Header'
import { mockNextUseRouter } from '../utils/mock-router'

describe('Component => Header', () => {
  mockNextUseRouter({
    route: '/',
    pathname: '/',
    query: '',
    asPath: '/',
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<Header />, {})
    const firstRender = asFragment()

    expect(firstRender).toMatchSnapshot()
  })
})
