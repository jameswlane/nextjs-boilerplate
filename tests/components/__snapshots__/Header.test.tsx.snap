// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component => Header matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <nav
      class="navbar navbar-expand-md navbar-dark fixed-top bg-dark"
    >
      <a
        class="navbar-brand"
        data-testid="top-nav-brand"
        href="/"
      >
        Next.js Boilerplate
      </a>
      <div
        class="collapse navbar-collapse"
      >
        <ul
          class="navbar-nav mr-auto"
        >
          <li
            class="nav-item active"
          >
            <a
              class="nav-link"
              data-testid="top-nav-home"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              data-testid="top-nav-about"
              href="/about"
            >
              About
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              data-testid="top-nav-contact"
              href="/contact"
            >
              Contact
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
